src/PropertyInfo/Extractors/PhpDocExtractor.php

```diff
@@ -102,7 +102,8 @@ public function extractTypes(\ReflectionProperty $reflectionProperty)
      */
     private function getFileReflector(\ReflectionClass $reflectionClass)
     {
-        if (!($fileName = $reflectionClass->getFileName())) {
+        if (!($fileName = $reflectionClass->getFileName())
+        or !in_array(pathinfo($fileName, PATHINFO_EXTENSION), ['php', 'phtml'])) {
```

Comment:
Replace `or` by `||` (not the same precedence, can lead to bugs).

It is better to test explicitly on hack type IMO because php files can have a lot of other extensions (`.php3`, `.php4`, `.php5`...).